src/test/java/org/broadinstitute/hellbender/tools/IndexFeatureFileIntegrationTest.java

```diff
@@ -0,0 +1,248 @@
+        for (int chr = 1; chr <= 4; chr++) {  //note: unusual loop
+            Assert.assertTrue(index.containsChromosome(String.valueOf(chr)));
+        }
+        for (int chr = 5; chr <= 22; chr++) { //note: unusual loop
+            Assert.assertFalse(index.containsChromosome(String.valueOf(chr)));
+        }
+        for (final String chr : Arrays.asList("X", "Y", "MT")){
+            Assert.assertFalse(index.containsChromosome(chr));
+        }
+
+        Assert.assertEquals(index.getSequenceNames(), Arrays.asList("1", "2", "3", "4"));
+    }
+
+    @Test
+    public void testVCFIndex_inferredName() {
+        final File ORIG_FILE = new File(getTestDataDir(), "test_variants_for_index.vcf");
+
+        final String[] args = new String[]{
+                "--feature_file" ,  ORIG_FILE.getAbsolutePath(),
+        };
+        final Object res = this.runCommandLine(args);
+        final File tribbleIndex = Tribble.indexFile(ORIG_FILE);
+        Assert.assertEquals(res, tribbleIndex.getAbsolutePath());
+        tribbleIndex.deleteOnExit();
+
+        final Index index = IndexFactory.loadIndex(res.toString());
+        Assert.assertTrue(index instanceof LinearIndex);
+        for (int chr = 1; chr <= 4; chr++) {  //note: unusual loop
+            Assert.assertTrue(index.containsChromosome(String.valueOf(chr)));
+        }
+        for (int chr = 5; chr <= 22; chr++) { //note: unusual loop
+            Assert.assertFalse(index.containsChromosome(String.valueOf(chr)));
+        }
+        for (final String chr : Arrays.asList("X", "Y", "MT")){
+            Assert.assertFalse(index.containsChromosome(chr));
+        }
+
+        Assert.assertEquals(index.getSequenceNames(), Arrays.asList("1", "2", "3", "4"));
+    }
+
+    @Test
+    public void testVCFGZIndex() {
+        final File ORIG_FILE = new File(getTestDataDir(), "test_variants_for_index.vcf.gz");
+        final File outName = BaseTest.createTempFile("test_variants_for_index.", ".idx");
+
+        final String[] args = new String[]{
+                "--feature_file" ,  ORIG_FILE.getAbsolutePath(),
+                "-O" ,  outName.getAbsolutePath()
+        };
+        final Object res = this.runCommandLine(args);
+        Assert.assertEquals(res, outName.getAbsolutePath());
+
+        final Index index = IndexFactory.loadIndex(res.toString());
+        Assert.assertTrue(index instanceof TabixIndex);
+        for (int chr = 1; chr <= 4; chr++) {  //note: unusual loop
+            Assert.assertTrue(index.containsChromosome(String.valueOf(chr)));
+        }
+        for (int chr = 5; chr <= 22; chr++) { //note: unusual loop
+            Assert.assertFalse(index.containsChromosome(String.valueOf(chr)));
+        }
+        for (final String chr : Arrays.asList("X", "Y", "MT")){
+            Assert.assertFalse(index.containsChromosome(chr));
+        }
+
+        Assert.assertEquals(index.getSequenceNames(), Arrays.asList("1", "2", "3", "4"));
+    }
+
+    @Test(enabled = false)
```

Comment:
Wait, why is this test disabled, but `testVCFGZIndex()` is enabled and passes tests? Why does only the "inferred name" case fail? Is it just a file-extension issue? (eg., when you don't specify the index file name explicitly, it chooses a `.tbi` extension, and we somehow have a problem with that?)